renku_notebooks/api/schemas/servers_get.py

```diff
@@ -305,20 +305,24 @@ def get_resource_requests(server):
             #   manage-compute-resources-container/#how-pods-with-resource-limits-are-run
             resources = {}
             if "cpu_request" in server_options_keys:
-                resources["cpu"] = server_options["cpu_request"]
+                resources["cpu"] = CpuField().deserialize(server_options["cpu_request"])
             if "mem_request" in server_options_keys:
-                resources["memory"] = float(server_options["mem_request"])
+                resources["memory"] = MemoryField().deserialize(
+                    server_options["mem_request"]
+                )
             if (
                 "disk_request" in server_options_keys
                 and server_options["disk_request"] is not None
                 and server_options["disk_request"] != ""
             ):
-                resources["storage"] = float(server_options["disk_request"])
+                resources["storage"] = MemoryField().deserialize(
+                    server_options["disk_request"]
+                )
             if (
                 "gpu_request" in server_options_keys
                 and int(server_options["gpu_request"]) > 0
```

Comment:
🤦 thanks Ralf I completely missed that... It worked because the gpu requests are always integers and in the worst case they are strings but with well-formed integer values... Nevertheless it was asking for trouble with that code.